overlord/snapstate/backend/snapdata.go

```diff
@@ -72,9 +92,6 @@ func removeDirs(dirs []string) error {
 		if err := os.RemoveAll(dir); err != nil {
 			return err
 		}
-
-		// Attempt to remove the parent directory as well (ignore any failure)
-		os.Remove(filepath.Dir(dir))
```

Comment:
Iirc this was used so that removing the last /snap/foo/123 revision would result in /snap/foo going away. Is this still accounted for somewhere? It sounds like we should have a spread test for this (to ensure that dirs are cleaned up if no snaps are installed anymore). Not sure if we have one right now.

Reply:
There's a spread test included in the PR. It does installation, removal and checks whether relevant directories exist. Although it already uses snaps with and without instance key, there's no case for instance-key-less snap being the sole one installed. I can extend it as needed.

Reply:
Extended the spread test to cover this too.

Reply:
are we sure this accounted for the undo cases,  we added explicit new Remove* calls for UndoSetupSnap but I don't see the equivalent for UndoCopySnapData (which would use removeDirs doing this remove before)

Reply:
also this bit was used by RemoveSnapCommonData  that didn't go a new remove

Reply:
As you noted the cleanup was missing from undoCopySnapData and doCopySnapData error path. Fixed that and added a spread test to cover this too.